### PR TITLE
Allow extension of vanilla themes

### LINKED ISSUE #18: Extending the vanilla themes
I thought about it again: It makes sense to allow users to add their own buildings to euro/int theme (by adding them to the config).

It is the easiest way to use custom assets with this mod.

For example, users could add these buildings to the european theme:
http://steamcommunity.com/sharedfiles/filedetails/?id=446190120
http://steamcommunity.com/sharedfiles/filedetails/?id=446201704

And all of these to the international theme:
http://steamcommunity.com/sharedfiles/filedetails/?id=423659623

To make it futureproof, vanilla buildings should be never be saved in the config (`isBuiltIn` field for `Configuration.Building`)

### PR DESCRIPTION
**Pull request by boformer**
Fixes #18

### CODE REVIEW

File: BuildingThemes/Configuration.cs [hunk 1]
```diff
@@ -121,10 +143,36 @@ public static void Serialize(string filename, Configuration config)
             }
         }
 
-        public static Configuration.Theme[] GetBuitInThemes()
+        public static void addBuiltInEuropeanTheme(Configuration config)
```
Comment (bloodypenguin) [inline BuildingThemes/Configuration.cs:139]: You can generify these two methods.

Comment (bloodypenguin) [inline BuildingThemes/Configuration.cs:139]: Just add a private helper method with 2 arguments

**Comment by bloodypenguin** [comment]
You should change Serialize/Deserialize methods too like we discussed (don't save built in buildings if they belong to built in theme and save them otherwise)

**Comment by boformer** [comment]
Testing with this configuration:

```
<?xml version="1.0" encoding="utf-8"?>
<Configuration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <Themes>
    <Theme name="European"> 
      <Buildings>
        <Building name="455335791.ERHL1A_Data" />
        <Building name="455719510.ERHL1B_Data" />
        <Building name="456227170.ERHL1C_Data" />
        <Building name="457264872.ERHL1Cnr1_Data" />
        <Building name="458977545.ERHL1Cnr Pub_Data" />
      </Buildings>
    </Theme>
    <Theme name="Edinburgh"> 
      <Buildings>
        <Building name="455335791.ERHL1A_Data" />
        <Building name="455719510.ERHL1B_Data" />
        <Building name="456227170.ERHL1C_Data" />
        <Building name="457264872.ERHL1Cnr1_Data" />
        <Building name="458977545.ERHL1Cnr Pub_Data" />
      </Buildings>
    </Theme>
  </Themes>
</Configuration>
```

Custom assets used: http://steamcommunity.com/workshop/filedetails/?id=455727807

**Comment by boformer** [comment]
It works.

**Comment by boformer** [comment]
@earalov I changed the Serialize method.

**Comment by boformer** [comment]
I am gone for 2 hours (cooking dinner). If you want, push the changes you want to make to this branch and merge it.

**Comment by bloodypenguin** [comment]
> I changed the Serialize method.

I see. Extended parts of built-in themes should be saved too.

**Comment by boformer** [comment]
That makes the isBuiltIn flag for themes kind of pointless. Can I just remove it?

We can either save all themes to the config file, or only those with at least one custom building.

**Comment by bloodypenguin** [comment]
No it's not pointless. I'm going to commit to this branch in several minutes and I will show why.

**Comment by boformer** [comment]
Great, i forgot to merge this :/

**Pull request merged**